apps/examples/security_test/security_test_main.c

```diff
@@ -0,0 +1,1582 @@
+
+unsigned int TC_see_read_write_secure_storage(void)
+{
+	int j = 0;
+	int ret = 0;
+	unsigned int test_result = 0;
+	unsigned char input[300];
+	unsigned int inlen[3] = { 18, SEE_MAX_DATA_SIZE, SEE_MAX_DATA_SIZE + 1 };
+	unsigned char output[SEE_MAX_DATA_SIZE];
+
+	printf("--------------------------------------------\n");
+	printf("  TESTCASE : see_read/write_secure_storage\n");
+	printf("--------------------------------------------\n");
+
+	memset(input, 1, 300);
+
+	for (j = 0; j < SSTORAGE_DATA_SLOT_INDEX_MAX; j++) {
+		see_selfprintf("[%d] ", j);
+
+		/* positive case 1 */
+		memset(output, 0, SEE_MAX_DATA_SIZE);
+		ret = see_write_secure_storage(input, inlen[0], j);
+		ret += see_read_secure_storage(output, &inlen[0], j);
+		if (ret || memcmp(input, output, inlen[0])) {
+			test_result |= 0x1;
+			see_selfprintf(" fail\n");
+			break;
+		}
+
+		/* positive case 2 */
+		memset(output, 0, SEE_MAX_DATA_SIZE);
+		ret = see_write_secure_storage(input, inlen[1], j);
+		ret += see_read_secure_storage(output, &inlen[1], j);
+		if (ret || memcmp(input, output, inlen[1])) {
+			test_result |= 0x1;
+			see_selfprintf(" fail\n");
+			break;
+		}
+
+		/* negative case 1 : exceed max size */
+		memset(output, 0, SEE_MAX_DATA_SIZE);
+		ret = see_write_secure_storage(input, inlen[2], j);
+		ret += see_read_secure_storage(output, &inlen[2], j);
+		if (!ret) {
+			test_result |= 0x1;
+			see_selfprintf(" fail\n");
+			break;
+		} else {
+			see_selfprintf(" success\n");
+		}
+
+		inlen[0] = 18;
+		inlen[1] = SEE_MAX_DATA_SIZE;
+		inlen[2] = SEE_MAX_DATA_SIZE + 1;
+	}
+
+	see_selfprintf("[%d] ", j);
+
+	ret = 0;
+	memset(output, 0, 200);
+	ret = see_write_secure_storage(input, inlen[0], SSTORAGE_DATA_SLOT_INDEX_MAX);
+	ret += see_read_secure_storage(output, &inlen[0], SSTORAGE_DATA_SLOT_INDEX_MAX);
+	if (!ret) {
+		test_result |= 0x1;
+		see_selfprintf(" fail\n");
+	} else {
+		see_selfprintf(" success\n");
+	}
+
+	return test_result;
+}
+
+#define SEE_GENERATE_KEY_TC 16
+
+unsigned int TC_see_generate_key(void)
+{
+	int ret = 0;
+	int i = 0;
+	int j = 0;
+	unsigned int test_result = 0;
+
+	unsigned int genkey_input[SEE_GENERATE_KEY_TC][3] = { {AES_KEY, 16, SEE_OK}, {AES_KEY, 24, SEE_OK}, {AES_KEY, 32, SEE_OK},
+		{AES_KEY, 48, SEE_OK}, {AES_KEY, 64, SEE_OK}, {AES_KEY, 80, SEE_ERROR},
+		{HMAC_KEY, 64, SEE_OK}, {HMAC_KEY, 128, SEE_OK}, {HMAC_KEY, 129, SEE_ERROR},
+		{ECC_KEY_BP256, 0, SEE_OK}, {ECC_KEY_NIST192, 0, SEE_OK},
+		{ECC_KEY_NIST224, 0, SEE_OK}, {ECC_KEY_NIST256, 0, SEE_OK},
+		{ECC_KEY_NIST384, 0, SEE_OK}, {ECC_KEY_NIST512, 0, SEE_OK},
+		{0xffff, 0, SEE_INVALID_INPUT_PARAMS}
+	};
+
+	printf("----------------------------------------\n");
+	printf("TESTCASE : see_generate_key\n");
+	printf("----------------------------------------\n");
+
+	for (j = 0; j < SEE_GENERATE_KEY_TC; j++) {
+		for (i = 0; i < 9; i++) {
+			see_selfprintf("[%d] ", j * 9 + i);
+			ret = see_generate_key(genkey_input[j][0], i, genkey_input[j][1], 0);
+			if (i == 8 && ret) {
```

Comment:
1. You had better check result first and then second condition, it is good habit to check state step by step
if (ret && i == 8)
2. If only last case is true(ret && i == 8), it is success case?

Reply:
I separated negative test cases (i=8) and removed if (ret && i == 8).